src/server/scripts/Northrend/IsleOfConquest/boss_ioc_horde_alliance.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "BattlegroundIC.h"
+
+enum BossSpells
+{
+    SPELL_BRUTAL_STRIKE       = 58460,
+    SPELL_DAGGER_THROW        = 67280,
+    SPELL_CRUSHING_LEAP       = 68506,
+    SPELL_RAGE                = 66776
+};
+
+enum BossEvents
+{
+    EVENT_BRUTAL_STRIKE       = 1,
+    EVENT_DAGGER_THROW        = 2,
+    EVENT_CRUSHING_LEAP       = 3,
+    EVENT_CHECK_RANGE         = 4
+};
+
+class boss_ioc_horde_alliance : public CreatureScript
+{
+public:
+    boss_ioc_horde_alliance() : CreatureScript("boss_ioc_horde_alliance") { }
+
+    struct boss_ioc_horde_allianceAI : public ScriptedAI
+    {
+        boss_ioc_horde_allianceAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void Reset() override
+        {
+            uint32 NPC_GUARD;
+            me->GetEntry() == NPC_HIGH_COMMANDER_HALFORD_WYRMBANE ? NPC_GUARD = NPC_SEVEN_TH_LEGION_INFANTRY : NPC_GUARD = NPC_KOR_KRON_GUARD;
```

Comment:
please use if/else instead of this syntax to initialize a local variable and follow the TC guidelines about variable namings, uppercase-only are defines/enums/macros